### PR TITLE
Update CAT detail view to support display of credit seats

### PR DESCRIPTION
**Pull request by rlucioni**
XCOM-510. Here's a screenshot of how this appears for a course with two credit seats:

@clintonb and/or @jimabramson, please review when able.

### CODE REVIEW

**Comment by rlucioni** [comment]
I'll be adding tests for the course detail view changes.

**Comment by rlucioni** [comment]
Rudimentary test added.

File: ecommerce/static/js/test/specs/utils/course_utils_spec.js [hunk 1]
```diff
@@ -41,5 +41,33 @@ define([
...
+                            input: [honor, verified, verified, professional],
+                            output: {
+                                filteredSeats: [verified, verified],
+                                residualSeats: [honor, professional]
+                            }
+                        },
+                    ];
+
+                _.each(data, function (data) {
+                    expect(CourseUtils.filterSeats(data.input, 'verified')).toEqual(data.output);
```
Comment (jimabramson) [inline ecommerce/static/js/test/specs/utils/course_utils_spec.js]: check residualSeats too?

**Comment by jimabramson** [comment]
@rlucioni LGTM apart from one question on the js test

File: ecommerce/static/js/utils/course_utils.js [hunk 2]
```diff
@@ -55,7 +55,37 @@ define([
...
+             * @param {String} seatType
+             * @returns {Object}
+             */
+            filterSeats: function (seats, seatType) {
+                var i,
+                    seat,
+                    filteredSeats = [],
+                    residualSeats = [];
+
+                for (i = 0; i < seats.length; i++) {
```
Comment (clintonb) [inline ecommerce/static/js/utils/course_utils.js]: [`_.groupBy`](http://underscorejs.org/#groupBy) can do this for you.

File: ecommerce/static/templates/_course_credit_seats.html [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                                print(moment.utc(expires).format('lll z'));
+                            }
+                        %></td>
+                    </tr>
+                <% }); %>
+            </tbody>
+        </table>
+    </div>
+
+    <div class="col-sm-4 seat-certificate-type">
```
Comment (clintonb) [inline ecommerce/static/templates/_course_credit_seats.html]: Maintain the consistency of certificate type being in the middle by moving this block above the table block, and [utilizing offsets](http://getbootstrap.com/css/#grid-offsetting) to keep it on its own row:

```
<div class="col-sm-4 col-sm-offset-4 seat-certificate-type">...</div>
<div class="col-sm-4"></div>
<div class="col-md-8">...</div>
```

**Comment by clintonb** [comment]
:+1:  (see notes about groupby and columns)

File: ecommerce/static/js/test/specs/utils/course_utils_spec.js (hunk 1 above)
Comment (rlucioni) [inline ecommerce/static/js/test/specs/utils/course_utils_spec.js]: @jimabramson both `filteredSeats` and `residualSeats` are checked. The utility returns an object with keys for both, and the test data includes the expected object.

File: ecommerce/static/js/utils/course_utils.js (hunk 2 above)
Comment (rlucioni) [inline ecommerce/static/js/utils/course_utils.js]: Mind blown. Thanks for the tip. Updated.

File: ecommerce/static/templates/_course_credit_seats.html (hunk 3 above)
Comment (rlucioni) [inline ecommerce/static/templates/_course_credit_seats.html]: Updated.

**Comment by rlucioni** [comment]
Updated screenshot:

**Pull request merged**